File: YunChat-Android/app/src/main/kotlin/com/android/yunchat/service/MessageService.kt
package com.android.yunchat.service

import com.android.yunchat.screen.chat.view.ChatViewModel

interface MessageService {
    fun sendTextMessage(content: String, chatViewModel: ChatViewModel)
    fun sendFileMessage(content: String, chatViewModel: ChatViewModel)
    fun sendImageMessage(content: String, chatViewModel: ChatViewModel)
    fun sendVideoMessage(content: String, chatViewModel: ChatViewModel)
    fun sendAgentMessage(content: String, chatViewModel: ChatViewModel)
    fun getMessageType(
        length: Int,
        type: String,
        content: String
    ): String
}

File: YunChat-Client/src/service/MessageService.ts
import { groupInfoType, messageType, restfulType } from "../../types";

export interface MessageService {
    getMessageType(type: string): string;
    getGroupMessageType(type: string, content: string)
    getNewMessageType( length: number, type: string, content: string): string
    sendTextMessage(content: string, callback?: (response: restfulType<messageType>) => void): void;
    sendFileMessage(content: string, callback?: (response: restfulType<messageType>) => void): void;
    sendImageMessage(content: string, callback?: (response: restfulType<messageType>) => void): void;
    sendVideoMessage(content: string, callback?: (response: restfulType<messageType>) => void): void;
    sendAgentMessage(content: string, callback?: (response: restfulType<messageType>) => void): void;
    sendGroupInvitedMessage(group: groupInfoType, callback?: (response: restfulType<messageType>) => void): void;
}